Создал классы c описанием страниц, в которых хранятся локаторы и методы:

Vezaris opened the pull request:
1. DataEntryPageScooter для страницы с вводом данных пользователя.
2. HomePageScooter для главной страницы сайта.
3. RentalPageScooter для страницы с параметрами заказа.

Создал класс EnvConfig, в котором хранится время для ждунов и url сайта для теста.

Создал класс DriverRule с описанием драйвера браузера.

Создал классы с тестами:
1. TestFaq для проверки текста в FAQ.
2. TestOrder для проверки флоу оформления заказа.

Vezaris committed:
Создал классы c описанием страниц, в которых хранятся локаторы и методы:
1. DataEntryPageScooter для страницы с вводом данных пользователя.
2. HomePageScooter для главной страницы сайта.
3. RentalPageScooter для страницы с параметрами заказа.

Создал класс EnvConfig, в котором хранится время для ждунов и url сайта для теста.

Создал класс DriverRule с описанием драйвера браузера.

Создал классы с тестами:
1. TestFaq для проверки текста в FAQ.
2. TestOrder для проверки флоу оформелния заказа.

nikolay-977 commented on src/main/java/Praktikum/EnvConfig.java:
```diff
@@ -0,0 +1,12 @@
+package Praktikum;
```
Необходимо исправить наименование пакетов в соответствии с правилами нейминга. Пакеты — допустимы только строчные буквы. В случае, если слов несколько, их следует разделять точкой. Префикс должен быть одним из доменных имен верхнего уровня, таких как com, ru и т.д. Наименование пакета должно отражать суть содержимого.

nikolay-977 commented on src/main/java/Praktikum/EnvConfig.java:
```diff
@@ -0,0 +1,12 @@
+public class EnvConfig {
+
+    /*---------------------------------------- Время для ждунов ---------------------------------------------*/
+    public static final int defoultTimeout = 5;
```
Необходимо исправить наименование констант в соответствии с правилами нейминга. Константы — состоят из заглавных символов. Если в имени несколько слов, они разделяются нижним подчёркиванием `_`.

nikolay-977 commented on src/main/java/Praktikum/Page/HomePageScooter.java:
```diff
@@ -0,0 +1,91 @@
+    // Верхняя кнопка заказа
+    private By topButtonOrderClick = By.xpath(".//button[@class='Button_Button__ra12g']");
+    // Нижняя кнопка заказа
+    private By downButtonOrderClick = By.xpath("//body/div/div/div[1]/div[4]/div[2]/div[5]/button");
```
Необходимо исправить: Локаторы не следует описывать через цепочку div элементов с индексами. Такие локаторы усложняют поддержку. Хороший локатор должен быть понятным и устойчив к изменениям. 

Вот пример хорошего локатора:

```
private By missClickButton = By.css("Order_Header__BZXOb");
```

Также можно использовать xpath. Xpath позволяет искать веб-элементы по любому атрибуту. Это хорошо описано в разделе UI-тестирование > Selenium > Поиск по HTML:

```
private By inputFieldName = By.xpath(".//input[@placeholder='* Имя']");
```

nikolay-977 commented on src/main/java/Praktikum/Page/HomePageScooter.java:
```diff
@@ -0,0 +1,91 @@
+    // Нижняя кнопка заказа
+    private By downButtonOrderClick = By.xpath("//body/div/div/div[1]/div[4]/div[2]/div[5]/button");
+
+
```
Можно улучшить: лишних пропусков строк следует избегать. Для зрительного разделения кода достаточно использовать один пропуск строки. Пропуски строк должны быть однообразными.

nikolay-977 commented on src/main/java/Praktikum/Page/RentalPageScooter.java:
```diff
@@ -0,0 +1,100 @@
+    /*----------------------------------------- ЛОКАТОРЫ ---------------------------------------------*/
+
+    // Поле даты доставки
+    private By deliveryDate = By.xpath("//input[@placeholder='* Когда привезти самокат']");
```
Необходимо исправить: В начале класса следует описывать переменные, а затем конструкторы и методы. Также можно установить курсор на корневую директорию и нажать сочетание клавиш Ctrl+Alt+L (Option+Command+L для Mac), затем выбрать checkbox rearrange entries и нажать ok.

nikolay-977 commented on src/test/java/Praktikum/TestOrder.java:
```diff
@@ -0,0 +1,109 @@
+    @Rule
+    public DriverRule driverRule = new DriverRule();
+
+    // Тестовые данные для оформления заказа
```
Можно улучшить: В параметризованных тестах для аннотации `@Parameterized.Parameters` лучше использовать аргумент name: `@Parameterized.Parameters(name = "Стоимость булочки. Тестовые данные: {0} {1}")`, где {0}, {1} - индексы параметров. С аргументом name наименование теста отобразится в отчете. А использование индексов параметров повысит информативность проверки.

nikolay-977 commented on src/test/java/Praktikum/TestFaq.java:
```diff
@@ -0,0 +1,36 @@
+        assertEquals("Не совпадает количество существующих и ожидаемых ответов", listOfFaqActual.length, listOfFaqExpected.length);
+        // Сравнение текста в ответах
+        for (int i = 0; i < listOfFaqActual.length; i++) {
+            assertEquals("Не совпадает текст в ответе номер " + i, listOfFaqActual[i], listOfFaqExpected[i]);
```
Необходимо исправить: Проверки в тестовых методах должны быть в одном стиле. Когда код написан в одном стиле, это улучшает читаемость кода, сокращает время локализации дефекта и упрощает его поддержку. Не следует в тестовых методах одновременно использовать проверки из разных фреймворков org.hamcrest.MatcherAssert(метод assertThat) и org.junit.Assert(методы assertEquals, assertTrue и т.д.). В этих фреймворках методы похожие, но имеют разные сигнатуры. Есть вероятность допустить ошибку.

nikolay-977 reviewed (commented):
Можно улучшить: хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройка (если необходима) и как запускать (`mvn clean test` - кавычки позволяют выполнять запуск из файла README.md). Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

Vezaris committed:
Исправлены замечания по ревью:
1. Добавлено описание в README
2. Переименованы пакеты
3. Переименованы константы в соответствии с правилами нейминга.
4. Изменен локатор, состоящий из цепочки div элементов. Теперь стал менее громоздким.
5. Убраны лишние пропуски строк.
6. Изменено расположение переменных. Теперь они первые описываются в классе.
7. Приведены к единообразию проверки в тестах. Убран assertEquals.

Vezaris committed:
Доббавлен пропущенный README файл.
